File: GlobalDevelopers.WeatherForecast/GlobalDevelopers.WeatherForecast/Services/WeatherService.cs
using System.Net;

namespace GlobalDevelopers.WeatherForecast.Services
{
    public class WeatherService : IWeatherService
    {
        public string GetWeatherData()
        {
            var url = "http://api.openweathermap.org/data/2.5/forecast/?q=London,gbr&units=metric&APPID=ea37e0ed896b17246d2c035325a0b1db";

            var client = new WebClient();

            var data = client.DownloadString(url);

            return data;
        }
    }
}

File: GlobalDevelopers.WeatherForecast/GlobalDevelopers.WeatherForecast/Scripts/Weather/WeatherService.js
WeatherForecast.service("WeatherService", function($http, $q) {
    var service = {};

    function httpPromise(url) {
        var deferred = $q.defer();

        $http.get(url)
            .then(function(data) {
                deferred.resolve(data);
            });

        return deferred.promise;
    };

    service.get = function() {
        return httpPromise('api/Weather');
    };

    return service;
})